commands/setUser.js

```diff
@@ -349,4 +349,124 @@ const updateUserNickname = async function(userid, nickname, server) {
+        WHERE
+            username = ? AND
+            discriminator = ?
+        `,
+        [username, discriminator]
+      );
+      if (!row) return [null, 'Invalid/unverified user.'];
+      assigneeId = row.userid;
+    } catch (e) {
+      console.error(e.toString());
+      return [{ message: e.toString() }, null];
+    } finally {
+      await db.close();
+    }
+  } else {
+    if (!assigneeInfo.match('([0-9])+')) {
+      return [
+        null,
+        'Invalid user format. Please supply either a user in <username>#<discriminator> format or a user ID'
+      ];
+    }
+    assigneeId = assigneeInfo;
+  }
+
+  // Get assignee GuildMember object
+  let assignee;
+  try {
+    assignee = await server.members.fetch(assigneeId);
+  } catch(e) {
+    console.error(e);
+    return [{ message: e.toString() }, 'Invalid/unverified user.'];
+  }
+
+  // Determine assigner's committee
+  let assignerCommittee;
+  if (assigner.roles.cache.some(role => role.name === 'PVP')) {
+    // PVP assigns Board officers
+    assignerCommittee = 'Board';
+  } else {
+    for (const committee of config.committees_list) {
+      if (assigner.roles.cache.some(role => role.name === `${committee} Officer`)) {
+        assignerCommittee = committee;
+        break;
+      }
+    }
+    if (!assignerCommittee) {
+      return [
+        null,
+        'Action not permitted. If you are a committee president, make sure that you also have ' +
+        'a committee officer role before running this command.'
+      ];
+    }
+  }
+  // Officer roles to be assigned based on assigner's committee
+  const acm_officer_role = server.roles.cache.find(role => role.name === 'ACM Officer');
+  const comm_officer_role = server.roles.cache.find(role => role.name === `${assignerCommittee} Officer`);
+
+  // Toggle officer roles
+  if (assignee.roles.cache.some(role => role.name === acm_officer_role.name)) {
+    if (assignee.roles.cache.some(role => role.name === comm_officer_role.name)) {
+      try {
+        await assignee.roles.remove([acm_officer_role, comm_officer_role]);
+        return [
+          null,
+          `Removed roles ${acm_officer_role.name} and ${comm_officer_role.name} from user ${assigneeInfo}.`
+        ];
+      } catch(e) {
+        return [{ message: e.toString() }, null];
+      }
+    } else {
+      return [null, 'Action not permitted. User is already an officer but is not in your committee.'];
+    }
+  } else {
+    if (!assignee.roles.cache.some(role => role.name === comm_officer_role.name)) {
+      try {
+        await assignee.roles.add([acm_officer_role, comm_officer_role]);
+        return [
+          null,
+          `Assigned roles ${acm_officer_role.name} and ${comm_officer_role.name} to user ${assigneeInfo}.`
+        ];
+      } catch(e) {
+        return [{ message: e.toString() }, null];
+      }
+    } else {
+      return [null, `Error: user has role ${comm_officer_role.name} but not ${acm_officer_role.name}.`];
```

Comment:
Here, I'd just say go ahead and add the ACM Officer role instead of returning an error